contracts/Plasma.sol

```diff
@@ -0,0 +1,211 @@
+pragma solidity 0.4.19;
+
+import 'zeppelin-solidity/contracts/ownership/Ownable.sol';
+import 'zeppelin-solidity/contracts/math/SafeMath.sol';
+import 'zeppelin-solidity/contracts/math/Math.sol';
+import './lib/MerkleProof.sol';
+import './lib/PriorityQueue.sol';
+import './Transaction.sol';
+
+import "./lib/RLP.sol";
+
+contract Plasma is Ownable {
+  mapping(address => bool) public operators;
+
+  modifier onlyOperator() {
+    if (operators[msg.sender]) _;
+  }
+
+  function setOperator(address _op, bool _status) public onlyOwner returns (bool success) {
+    operators[_op] = _status;
+    return true;
+  }
+
+  function isOperator() public view returns (bool success) {
+    return operators[msg.sender];
+  }
+
+  // blocks
+  struct Block {
+    bytes32 merkleRoot;
+    uint256 timestamp;
+  }
+
+  uint public blockCount;
+  mapping(uint => Block) public blocks;
+
+  event BlockSubmitted(uint indexed blockIndex, address operator, bytes32 indexed merkleRoot);
+
+  function submitBlock(bytes32 _merkleRoot, uint _blockIndex) public onlyOperator {
+    require(blockCount == _blockIndex);
+
+    Block memory newBlock = Block(_merkleRoot, block.timestamp);
+
+    blocks[_blockIndex] = newBlock;
+
+    BlockSubmitted(_blockIndex, msg.sender, _merkleRoot);
+
+    blockCount += 1;
+  }
+
+  // deposits
+  struct Deposit {
+    address depositor;
+    uint amount;
+  }
+
+  mapping(uint => Deposit) public deposits;
+  uint public depositCount;
+
+  event Deposited(uint indexed uid, address depositor, uint amount);
+
+  function deposit(address _currency, uint _amount) payable public {
+    require(_amount > 0);
+
+    if (_currency == address(0)) {
```

Comment:
Interesting way of support both ETH and ERC

Reply:
ERC is not supported yet but it should be easy to make it work. We have omitted that part for the sake of simplicity